lb: Use more neutral terminology

brb opened the pull request:
This PR replaces `svc master` by `svc frontend` and `slave` by `backend slot`.

tklauser commented on pkg/maps/lbmap/lbmap.go:
```diff
@@ -72,7 +72,7 @@ func (*LBBPFMap) UpsertService(
 		svcVal.SetBackendID(loadbalancer.BackendID(backendID))
 		svcVal.SetRevNat(int(svcID))
-		svcKey.SetSlave(slot) // TODO(brb) Rename to SetSlot
+		svcKey.SetEndpoint(slot) // TODO(brb) Rename to SetSlot
```
Remove the TODO? Or rename it to `SetSlot` as indicated?

qmonnet replied:
Should you take that chance to implement your `TODO` instead for this one?
[Edit] Oh, Tobias beat me on that one, sorry.

qmonnet reviewed (approved):
Looks great, thanks for that.

joestringer commented on bpf/lib/common.h:
```diff
@@ -572,7 +572,7 @@ struct ct_entry {
 	union v6addr address;	/* Service virtual IPv6 address */
 	__be16 dport;		/* L4 port filter, if unset, all ports apply */
-	__u16 slave;		/* Backend iterator, 0 indicates the master service */
+	__u16 endpoint;		/* Backend iterator, 0 indicates the primary service */
```
I note that here we call it `endpoint` (which admittedly is consistent with k8s terminology, but also collides with long-held Cilium terminology for the container network termination point), yet in the comment we clarify it by calling it the backend.

Did you consider frontend/backend terminology?

borkmann replied:
frontend/backend would actually be more clear, or, short `fe`/`be` if vars become too long.

aanm replied:
yeah I thought about the same the first time I saw this PR but I couldn't think on an alternative, `fe/be` seems a good approach to me as well.

brb replied:
Yep, frontend/backend makes sense. However, the particular field at the line 575 points to a "service slot" (`lb6_service`) which contains a backend_id pointing to a real backend. This is the main reason why I didn't want to use `backend` here. Maybe we should rename `slave` to `backend_slot` or `backend_no`?

joestringer replied:
if it contains a `backend_id`, is there anything stopping you from calling it `backend_id`? :-)

id/slot/no, any of these sound reasonable.

brb replied:
Unfortunately, it doesn't contain an ID: `lb6_key{.slave = 1 ... lb6_service[lb6_key{.slave = 0}].count} => lb6_service.backend_id => lb6_backend`. As you can see, `slave` refers to a "backend slot index" ;-) OK, I think I'm going to rename to `backend_slot`, and add a short comment about relations.

brb committed:
lb: Use more neutral terminology

This commit replaces "svc master" by "svc frontend" and "slave" by
"backend slot".

Signed-off-by: Martynas Pumputis <m@lambda.lt>
